soap/index.d.ts

```diff
@@ -19,12 +19,22 @@ export class ClientSSLSecurity implements Security {
     constructor(key: Buffer | string, cert: Buffer | string, defaults?: any);
 }
 
-interface Client extends events.EventEmitter {
-    setSecurity(s: Security): void;
-    [method: string]: (args: any, fn: (err: any, result: any) => void, options?: any, extraHeaders?: any) => void;
-    addSoapHeader(headJSON: any): void;
-    setEndpoint(endpoint: string): void;
+export interface Client extends events.EventEmitter {
+    [method: string]: any;
```

Comment:
Both `addBodyAttribute()` and `addSoapHeader()` were causing the error `not assignable to string index type` and I couldn't find a better way to avoid that.

The order doesn't appear to matter, so I sorted alphabetically.